### PR TITLE
Bug 1092427 - Text Selection for Dialer, CallScreen, and emergency diale...

### PR DESCRIPTION
**Pull request by tamarahills**
...r apps.

### CODE REVIEW

**Comment by try-server-hook** [comment]
[tamarahills tamarahills](https://github.com/tamarahills) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=5152f5042fef)

File: shared/style/dialer/dialer.css [hunk 1]
```diff
@@ -1,5 +1,12 @@
+* {
+  /* By default do not allow any user text selection, only enable for certain
+  parts on a per-element/region area */
+  -moz-user-select: none;
+}
+
 html {
   font-size: 10px;
+  -moz-user-select: none;
```
Comment (DouglasSherk) [inline shared/style/dialer/dialer.css]: Why is this needed?

File: shared/style/dialer/dialer.css [hunk 2]
```diff
@@ -9,6 +16,7 @@ html,body {
   margin: 0;
   color: white;
   background-color: white;
+  -moz-user-select: none;
```
Comment (DouglasSherk) [inline shared/style/dialer/dialer.css]: Why is this needed?

File: apps/communications/dialer/index.html [hunk 3]
```diff
@@ -167,7 +167,7 @@ <h1 data-l10n-id="callLog"></h1>
         <div id="keypad-panel" role="tabpanel" class="bb-tabpanel" aria-labelledby="option-keypad">
           <div id="phone-number-view-container" dir="ltr">
             <div class="phone-number-view-wrapper-input">
-              <input id="phone-number-view" type="text" class="phone-number-font font-light" readonly="readonly" data-l10n-id="phoneNumberInput">
+              <input id="phone-number-view" type="text" class="phone-number-font font-light textfield" readonly="readonly" data-l10n-id="phoneNumberInput">
```
Comment (DouglasSherk) [inline apps/communications/dialer/index.html]: Why is this needed?

File: apps/emergency-call/index.html [hunk 4]
```diff
@@ -53,7 +53,7 @@
       <section id="keyboard-view" class="full-height">
         <div id="phone-number-view-container">
           <div class="phone-number-view-wrapper-input">
-            <input id="phone-number-view" type="text" class="phone-number-font font-light" readonly="readonly" data-l10n-id="phoneNumberInput">
+            <input id="phone-number-view" type="text" class="phone-number-font font-light textfield" readonly="readonly" data-l10n-id="phoneNumberInput">
```
Comment (DouglasSherk) [inline apps/emergency-call/index.html]: Why is this needed?

File: apps/callscreen/style/oncall.css [hunk 5]
```diff
@@ -8,6 +8,7 @@
     font-size: 10px;
     color: white;
     background: transparent;
+    -moz-user-select: none;
```
Comment (DouglasSherk) [inline apps/callscreen/style/oncall.css]: Why do we need to put this under the ```*``` selector elsewhere but not here?

File: shared/style/dialer/dialer.css [hunk 6]
```diff
@@ -1,5 +1,12 @@
+* {
+  /* By default do not allow any user text selection, only enable for certain
+  parts on a per-element/region area */
+  -moz-user-select: none;
```
Comment (DouglasSherk) [inline shared/style/dialer/dialer.css]: We shouldn't need to do this because this should fall through if applied to the ```body``` or ```html``` elements.

File: apps/callscreen/style/oncall.css (hunk 5 above)
Comment (DouglasSherk) [inline apps/callscreen/style/oncall.css]: Try removing the ```html``` selector from here. If it works, please file a bug for it.

File: apps/communications/dialer/index.html (hunk 3 above)
Comment (DouglasSherk) [inline apps/communications/dialer/index.html]: We talked about this, no longer an issue.

File: apps/emergency-call/index.html (hunk 4 above)
Comment (DouglasSherk) [inline apps/emergency-call/index.html]: We talked about this, no longer an issue.

File: shared/style/dialer/dialer.css (hunk 2 above)
Comment (DouglasSherk) [inline shared/style/dialer/dialer.css]: This shouldn't be needed given https://github.com/mozilla-b2g/gaia/pull/27159/files#r22540359

File: shared/style/dialer/dialer.css (hunk 1 above)
Comment (DouglasSherk) [inline shared/style/dialer/dialer.css]: Maybe this should be applied on the ```body``` tag only. I don't really have a good justification here, but it makes more sense to me. The ```body``` tag contains only actual content, but the ```html``` tag contains lots of metadata (such as ```head``` content) which is not selectable to begin with.

File: shared/style/dialer/keypad.css [hunk 7]
```diff
@@ -46,9 +46,9 @@
...
   text-align: left;
   cursor: none;
-  -moz-user-select: none;
   padding: 0;
-  pointer-events: none;
+  pointer-events: auto;
+  -moz-user-select: text;
 }
 
 #fake-phone-number-view {
```
Comment (DouglasSherk) [inline shared/style/dialer/keypad.css]: Please open a bug to investigate removing this.

File: shared/style/dialer/dialer.css (hunk 6 above)
Comment (tamarahills) [inline shared/style/dialer/dialer.css]: Hi Doug,

I did some more testing on this and what i found was that I needed to set it so none for the *.  If I set it to none on the *, then it's not needed on the html or body.  Also, adding -moz-select-user: none to both the html, and the body (without adding it at the *) is not sufficient to keep unwanted elements from being selected.

File: apps/callscreen/style/oncall.css (hunk 5 above)
Comment (tamarahills) [inline apps/callscreen/style/oncall.css]: Just to follow up on this one, I removed the html selector and it did not work.  The page was completely distorted.

**Comment by try-server-hook** [comment]
[tamarahills tamarahills](https://github.com/tamarahills) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=ea8b2746b363)

**Comment by try-server-hook** [comment]
[tamarahills tamarahills](https://github.com/tamarahills) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8892ce5e0d67)
